girder/models/user.py

```diff
@@ -244,26 +249,43 @@ def createUser(self, login, password, firstName, lastName, email,
             'groupInvites': []
         }
 
-        self.setPassword(user, password)
+        self.setPassword(user, password, save=False)
+
+        self.setPublic(user, public, save=False)
+
+        # An id is required in order to grant the user access on itself, but
+        # since the user isn't fully created yet, it should not be saved
+        user['_id'] = ObjectId()
```

Comment:
Have you investigated to make sure this doesn't cause any weirdness as a result of pymongo conditioning behavior on the presence of an _id field? https://github.com/mher/pymongo/blob/master/pymongo/collection.py#L210-L215